src/utils/forward-events.ts

```diff
@@ -0,0 +1,90 @@
+import type {
+  JobCompletionEmit,
+  EventForwarder,
+  PenumbraErrorEmit,
+  ProgressEmit,
+} from '../types';
+
+const progressEventQueue: ProgressEmit[] = [];
+let progressEventQueueInit = false;
+const jobCompletionEventQueue: JobCompletionEmit[] = [];
+let jobCompletionEventQueueInit = false;
+const penumbraErrorEventQueue: PenumbraErrorEmit[] = [];
+let penumbraErrorEventQueueInit = false;
+const errorEventQueue: ErrorEvent[] = [];
+let errorEventQueueInit = false;
```

Comment:
Super optional: Would it be of use to combine some of these into an object, or multiple objects?